### PR TITLE
Issue632: pretty printing array brackets using common type

### PR DESCRIPTION
**Pull request by ftomassetti**
See #632

### CODE REVIEW

**Comment by matozoid** [comment]
Work in progress?

**Comment by ftomassetti** [comment]
Well, I guess I just need to fix the tests, not the code :) working on it

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java [hunk 1]
```diff
@@ -36,4 +39,12 @@ default VariableDeclarator getVariable(int i) {
     }
 
     N setVariables(NodeList<VariableDeclarator> variables);
+
+    public default Type getCommonType() {
+        return getVariables().stream().map(v -> v.getType())
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: You're conflicting with #631 :-O Shall I go pull some PR's?

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java [hunk 2]
```diff
@@ -36,4 +39,12 @@ default VariableDeclarator getVariable(int i) {
...
+                .sorted((Comparator<Type<?>>) (a, b) -> a.getArrayLevel() - b.getArrayLevel())
+                .findFirst()
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: I guess you're going to add some code that checks the annotations?

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java (hunk 1 above)
Comment (ftomassetti) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: I will wait for #631 to be merged and adapt

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java (hunk 2 above)
Comment (ftomassetti) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: Damned annotations :D
Well, this seems an interesting problem that I will fix :D

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: Oh look, I can edit eveybody's comments! :D

Comment (ftomassetti) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: Ah ah ah, with great power comes great responsibility :D 
Ok, I will do it, after #631 is merged

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java (hunk 1 above)
Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: Okay, it's merged. Note that ".junit." has disappeared from the package name of many unit tests.

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java]: Oh hey, the merged getCommonType and yours are not the same. Mine only returns the type of the variables if it is the same for each one. What shall we do about naming?

The thing you are describing has no specific name in the JLS, as far as I know. Neither does mine.

**Comment by ftomassetti** [comment]
Ok, considering annotations was tricky...

**Comment by matozoid** [comment]
This is awesome!

*sits and cries in a corner* ;)

**Comment by ftomassetti** [comment]
Maybe not the clearest code ever written... :)
This is the day I deserved the hell of developers

**Comment by matozoid** [comment]
Oooh, I just thought of something. getMaximumCommonType() is only used for pretty printing, I would say. So why not move it to the pretty printer?

**Comment by ftomassetti** [comment]
I think it could be potentially be more useful, but given it is not a clear method it could make sense to move to a less visible/general place. I will do that

**Comment by matozoid** [comment]
It can always be moved around again.

**Comment by matozoid** [comment]
Thanks for doing this feature!

**Comment by ftomassetti** [comment]
thanks for the feedback

**Pull request merged**